libavformat/dashenc_http.c

```diff
@@ -313,7 +313,11 @@ static void *thr_io_close(connection *conn) {
     if (ret < 0 || response_code >= 500) {
         av_log(NULL, AV_LOG_INFO, "-event- request failed ret=%d, conn_nr: %d, response_code: %d, url: %s.\n", ret, conn->nr, response_code, conn->url);
         abort_if_needed(conn->must_succeed);
-        ff_format_io_close(conn->s, &conn->out);
+        //must check if this is NULL or it causes crash on segmentation fault due to NULL pointer
+        //check why conn->s (AVFormatContext) becomes NULL
```

Comment:
Can you check if this does not happen when you use the http_persistent=1 option?
If so we can remove this code and I will create a ticket for supporting non persistent connections.

Reply:
I'll check.

Reply:
I tested with http_persistent=1, it never gets segmentation fault or crashes.